src/Pelagos/Bundle/AppBundle/Command/ColdStorageFlagCommand.php

```diff
@@ -0,0 +1,181 @@
+                    }
+                    $output->writeln('original submission ID is: ' . $dataset->getDatasetSubmission()->getId());
+                    $output->writeln('new submission ID is initially: ' . $newDatasetSubmission->getId());
+
+                    // This method sets submission status to: DatasetSubmission::STATUS_IN_REVIEW (3).
+                    $newDatasetSubmission->setDatasetSubmissionReviewStatus();
+                    // Set filesize of original file in new submission.
+                    $newDatasetSubmission->setDatasetFileColdStorageArchiveSize($size);
+                    // Set hash of original file in new submission.
+                    $newDatasetSubmission->setDatasetFileColdStorageArchiveSha256Hash($hash);
+                    // Set the creator of this entity to the user known as 'system'.
+                    $newDatasetSubmission->setCreator($systemPerson);
+
+                    // Set options for a new replacement datafile of the supplied Cold-Storage stubfile.
+                    $newDatasetSubmission->setDatasetFileTransferStatus(DatasetSubmission::TRANSFER_STATUS_NONE);
+                    $newDatasetSubmission->setDatasetFileName(null);
+                    $newDatasetSubmission->setDatasetFileSize(null);
+                    $newDatasetSubmission->setDatasetFileSha256Hash(null);
+                    $newDatasetSubmission->setDatasetFileTransferType(DatasetSubmission::TRANSFER_TYPE_SFTP);
+                    $newDatasetSubmission->setDatasetFileUri($stubFileName);
+                    $newDatasetSubmission->setDistributionFormatName('Cold Storage archive - ' .
+                        $datasetSubmission->getDistributionFormatName());
+
+                    // This is normally null, unless manually set. Handle that gracefully.
+                    $newDatasetSubmissionId = $newDatasetSubmission->getId();
+                    $metadata = $entityManager->getClassMetaData(get_class($newDatasetSubmission));
+                    $newDatasetSubmissionIdGenerator = $metadata->idGenerator;
+                    if ($newDatasetSubmissionId !== null) {
+                        // Temporarily change the ID generator to AssignedGenerator.
+                        $metadata->setIdGenerator(new AssignedGenerator());
+                    }
+
+                    $entityManager->persist($newDatasetSubmission);
+                    $entityManager->flush($newDatasetSubmission);
+
+                    if ($newDatasetSubmissionId !== null) {
+                        $metadata->setIdGenerator($newDatasetSubmissionIdGenerator);
+                    }
+
+                    $dataset->setDatasetSubmission($newDatasetSubmission);
+                    $entityManager->persist($dataset);
+                    $entityManager->flush($dataset);
+
+                    // This is an ugly hack, having to re-persist, but there seems not to be a cleaner way...
```

Comment:
There is no setter for status in DatasetSubmission Entity, so i think we should probably not change it using Reflection class. I think lot of other attributes are tied to it, like submissionTimeStamp and submitter which also need to be set or else we would have lot of discrepancies between these datasets and others.

Reply:
I'm using reflections here just for changing it back to the value it was originally in the original dataset submission. The issue is that I cannot call dataset->setDatasetSubmission() on the new submission if the value is something other than DatasetSubmission :STATUS_COMPLETE or DatasetSubmission::STATUS_IN_REVIEW otherwise it'll be specifically nulled in the Dataset->setDatasetSubmission().

The constructor in DatasetSubmission.php handles submissionTimeStamp and submitter. modificationTimeStamp and modifier will get stomped, but everything that touches the entity stomps these anyway.

Reply:
I could have also used reflections to bypass dataset's setter. This might be cleaner.